hypervisor/arch/x86/guest/assign.c

```diff
@@ -213,7 +213,22 @@ ptirq_build_physical_rte(struct acrn_vm *vm, struct ptirq_remapping_info *entry)
 		}
 
 		/* update physical delivery mode, dest mode(logical) & vector */
-		vector = irq_to_vector(phys_irq);
+		if (is_lapic_pt_configured(vm)) {
+			/*
+			 * In local APIC passthrough case, when devices triggered a INTx interrupt, this interrupt
+			 * would be delivered to vCPU directly. For this case, need to set the virtual vector in
+			 * the 'Interrupt Vector' field of physical IOxAPIC I/O REDIRECTION TABLE REGISTER (bits 7:0)
+			 * and 'Vector' field of vt-d Interrupt Remapping Table Entry (IRTE) for Remapped Interrupts.
+			 *
+			 * Assumption:
+			 * (a) IOAPIC pins won't be shared between LAPIC PT guest and other guests;
+			 * (b) The guest would not trigger this IRQ before it switched to x2 APIC mode.
```

Comment:
Put this assumption to DOC side too.

Reply:
Will add the DOC once aligned with nanlin where to add this DOC.